Event/DoctrineBackendOptimizeListener.php

```diff
@@ -30,9 +28,9 @@ class DoctrineBackendOptimizeListener implements IterationListener
     protected $doctrine;
 
     /**
-     * @param RegistryInterface $doctrine
+     * @param ContainerAwareInterface $doctrine
      */
-    public function __construct(RegistryInterface $doctrine)
+    public function __construct(ContainerAwareInterface $doctrine)
```

Comment:
In the case of MongoDB driver $doctrine should be an instance of Doctrine\Bundle\MongoDBBundle\ManagerRegistry and in the case of ORM driver, it should be an instance of \Doctrine\Bundle\DoctrineBundle\Registry.For this reason I used ContainerAwareInterface!!